api/integrations/slack/views.py

```diff
@@ -19,40 +21,45 @@
 from integrations.slack.slack import SlackWrapper
 
 from .authentication import OauthInitAuthentication
-from .exceptions import FrontEndRedirectURLNotFound, InvalidStateError
+from .exceptions import (
+    FrontEndRedirectURLNotFound,
+    InvalidStateError,
+    SlackConfigurationDoesNotExist,
+)
+from .pagination import ChannelListPagination
 from .permissions import OauthInitPermission
 
 signer = TimestampSigner()
 
 
+class SlackGetChannelsViewSet(GenericViewSet):
+    serializer_class = SlackChannelListSerializer
+    pagination_class = ChannelListPagination
+
+    def get_queryset(self) -> SlackWrapper:
+        environment = Environment.objects.get(
+            api_key=self.kwargs["environment_api_key"]
+        )
+        try:
+            config = SlackConfiguration.objects.get(project=environment.project)
+        except ObjectDoesNotExist as e:
+            raise SlackConfigurationDoesNotExist() from e
+        slack_wrapper = SlackWrapper(api_token=config.api_token)
+
+        return slack_wrapper
+
+    def list(self, request, *args, **kwargs):
+        queryset = self.filter_queryset(self.get_queryset())
```

Comment:
I think adding `filter_queryset` is probably a bad idea here. The only filter backend we have is the `DjangoFilterBackend` which will likely throw some exception if invoked (which I think would only happen if someone accidentally added a `filterset_fields` attribute to this class but still).